web/src/features/exchanges/ExchangeArrow.tsx

```diff
@@ -26,124 +23,82 @@ function ExchangeArrow({
   map,
   colorBlindMode,
 }: ExchangeArrowProps) {
-  const mapZoom = map.getZoom();
-  const colorBlindModeEnabled = colorBlindMode;
-  const absFlow = Math.abs(data.netFlow ?? 0);
   const { co2intensity, lonlat, netFlow, rotation, key } = data;
-  const setIsMoving = useSetAtom(mapMovingAtom);
   if (!lonlat) {
     return null;
   }
 
-  useEffect(() => {
-    const cancelWheel = (event: Event) => event.preventDefault();
-    const exchangeLayer = document.querySelector('#exchange-layer');
-    if (!exchangeLayer) {
-      return;
-    }
-    exchangeLayer.addEventListener('wheel', cancelWheel, {
-      passive: false,
-    });
-    return () => exchangeLayer.removeEventListener('wheel', cancelWheel);
-  }, []);
-
-  const imageSource = useMemo(() => {
-    const prefix = colorBlindModeEnabled ? 'colorblind-' : '';
-    const intensity = quantizedCo2IntensityScale(co2intensity);
-    const speed = quantizedExchangeSpeedScale(Math.abs(netFlow));
-    return resolvePath(`images/arrows/${prefix}arrow-${intensity}-animated-${speed}`)
-      .pathname;
-  }, [colorBlindModeEnabled, co2intensity, netFlow]);
-
+  const absFlow = Math.abs(netFlow ?? 0);
+  // Don't render if the flow is very low ...
+  if (absFlow < 1) {
+    return null;
+  }
+  const mapZoom = map.getZoom();
   const projection = map.project(lonlat);
-
   const transform = {
     x: projection.x,
     y: projection.y,
     k: 0.04 + (mapZoom - 1.5) * 0.1,
     r: rotation + (netFlow > 0 ? 180 : 0),
   };
 
-  // Don't render if the flow is very low ...
-  if (absFlow < 1) {
-    return null;
-  }
-
-  // ... or if the arrow would be very tiny ...
-  if (transform.k < 0.1) {
-    return null;
-  }
-
-  // ... or if it would be rendered outside of viewport.
-  if (transform.x + 100 * transform.k < 0) {
+  if (
+    // or if the arrow would be very tiny
+    transform.k < 0.1 ||
+    // or if it would be rendered outside of viewport.
+    transform.x + 100 * transform.k < 0 ||
+    transform.x - 100 * transform.k > viewportWidth ||
+    transform.y - 100 * transform.k > viewportHeight
+  ) {
```

Comment:
Why do we not check for `absFlow < 1` and `transform.y + 100 * transform.k < 0` anymore?

Reply:
Abs flow is checked further up so we exit sooner and avoid functions calls that we won't use anyway.

The other check is a miss on my side so I'll add that back 🙂